app/helpers/archives_helper.rb

```diff
@@ -15,6 +15,17 @@ def status_explanation(archive)
     end
   end
 
+  def archive_delete_link(archive)
+    link_to(
+      content_tag(:span, "", class: %w{glyphicon glyphicon-remove}) + " Delete Archive",
+      archive,
+      class: %w{btn btn-sm btn-danger},
+      method: :delete,
+      data: { confirm: t("archives.index.delete_archive") },
+      title: t("archives.index.delete_archive")
+    ) unless archive.s3_url.nil?
```

Comment:
Would `if archive.current_state == :ready` work here ? It would cover the edge case where an archive is marked as "ready" but has failed to upload to S3 (and thus has a nil `s3_url` but still can be removed)